components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -987,8 +999,15 @@ private void updateMetadataForH2(Resource resource, String resourceTypeId, boole
                     template.executeInsert(UPDATE_RESOURCE_H2,
                             preparedStatement -> {
                                 int initialParameterIndex = 1;
-                                preparedStatement.setInt(initialParameterIndex,
-                                        PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
+                                int tenantId;
+                                String tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
+                                if (StringUtils.isBlank(tenantDomain)) {
+                                    tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                            .getTenantId();
+                                } else {
+                                    tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+                                }
```

Comment:
This logic seems to repeat. Can we move this logic to a common function / utility and re-use?